roachtest: sqlalchemy failed

roachtest.sqlalchemy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20412421?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20412421?buildTab=artifacts#/sqlalchemy) on release-25.2 @ [7af0f9c919e2d46a2fd70203957109a3a52bab35](https://github.com/cockroachdb/cockroach/commits/7af0f9c919e2d46a2fd70203957109a3a52bab35):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.5-dev-7af0f9c919e2d46a2fd70203957109a3a52bab35
Tests run against sqlalchemy 2.0.43
535 Total Tests Run
531 tests passed
4 tests failed
812 tests skipped
3 tests ignored
0 tests passed unexpectedly
4 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-MyCkConst] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-my_ck_const] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_no_constraint[True] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-None] (unexpected)
For a full summary look at the sqlalchemy artifacts 
An updated blocklist (sqlAlchemyBlocklist) is available in the artifacts' sqlalchemy log
test artifacts and logs in: /artifacts/sqlalchemy/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-20412421/sqlalchemy/1756535330747/1756535572751)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152727 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.2.5-rc]
- #152725 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.3.19-rc]
- #152722 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.23-rc]
- #152720 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.3.0-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlalchemy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54031

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.sqlalchemy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20413806?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20413806?buildTab=artifacts#/sqlalchemy) on release-25.2 @ [7af0f9c919e2d46a2fd70203957109a3a52bab35](https://github.com/cockroachdb/cockroach/commits/7af0f9c919e2d46a2fd70203957109a3a52bab35):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.5-dev-7af0f9c919e2d46a2fd70203957109a3a52bab35
Tests run against sqlalchemy 2.0.43
535 Total Tests Run
531 tests passed
4 tests failed
812 tests skipped
3 tests ignored
0 tests passed unexpectedly
4 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_no_constraint[True] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-None] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-my_ck_const] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-MyCkConst] (unexpected)
For a full summary look at the sqlalchemy artifacts 
An updated blocklist (sqlAlchemyBlocklist) is available in the artifacts' sqlalchemy log
test artifacts and logs in: /artifacts/sqlalchemy/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-20413806/sqlalchemy/1756622255182/1756622501580)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152780 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-master]
- #152779 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.3]
- #152778 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3]
- #152777 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.3.1-rc]
- #152775 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1]
- #152727 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.2.5-rc]
- #152725 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.3.19-rc]
- #152722 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.23-rc]
- #152720 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlalchemy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.sqlalchemy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20415850?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/20415850?buildTab=artifacts#/sqlalchemy) on release-25.2 @ [7af0f9c919e2d46a2fd70203957109a3a52bab35](https://github.com/cockroachdb/cockroach/commits/7af0f9c919e2d46a2fd70203957109a3a52bab35):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.5-dev-7af0f9c919e2d46a2fd70203957109a3a52bab35
Tests run against sqlalchemy 2.0.43
535 Total Tests Run
531 tests passed
4 tests failed
812 tests skipped
3 tests ignored
0 tests passed unexpectedly
4 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-None] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-MyCkConst] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_standalone[True-my_ck_const] (unexpected)
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTestExtra_cockroachdb+psycopg2_9_5_0::test_check_constraint_no_constraint[True] (unexpected)
For a full summary look at the sqlalchemy artifacts 
An updated blocklist (sqlAlchemyBlocklist) is available in the artifacts' sqlalchemy log
test artifacts and logs in: /artifacts/sqlalchemy/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-20415850/sqlalchemy/1756796295102/1756796555042)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152780 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-master]
- #152779 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.3]
- #152778 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.3]
- #152777 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.3.1-rc]
- #152775 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1]
- #152727 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.2.5-rc]
- #152725 roachtest: sqlalchemy failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.3.19-rc]
- #152722 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.23-rc]
- #152720 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-25.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlalchemy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/sqlalchemy.go
+++ b/pkg/cmd/roachtest/tests/sqlalchemy.go
@@ -128,6 +128,22 @@ func runSQLAlchemy(ctx context.Context, t test.Test, c cluster.Cluster) {
 		t.Fatal(err)
 	}
 
+	// The test suite requires two schemas to be created ahead of time.
+	// See https://github.com/sqlalchemy/sqlalchemy/blob/b3feb9cd3e87960ee65968b712e73d8464385a4a/README.unittests.rst?plain=1#L139.
+	db, err := c.ConnE(ctx, t.L(), node[0], option.DBName("defaultdb"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+	for _, cmd := range []string{
+		`CREATE SCHEMA IF NOT EXISTS test_schema`,
+		`CREATE SCHEMA IF NOT EXISTS test_schema_2`,
+	} {
+		if _, err := db.ExecContext(ctx, cmd); err != nil {
+			t.Fatal(err)
+		}
+	}
+
 	blocklistName, expectedFailures := "sqlAlchemyBlocklist", sqlAlchemyBlocklist
 	ignoredlistName, ignoredlist := "sqlAlchemyIgnoreList", sqlAlchemyIgnoreList
 	t.L().Printf("Running cockroach version %s, using blocklist %s, using ignoredlist %s",

